node/src/components/storage.rs

```diff
@@ -710,6 +705,73 @@ impl Storage {
             .transpose()
     }
 
+    /// Retrieves the highest block from the storage, if one exists.
+    /// May return an LMDB error.
+    fn get_highest_block<Tx: Transaction>(
+        &self,
+        txn: &mut Tx,
+    ) -> Result<Option<Block>, LmdbExtError> {
+        self.block_height_index
+            .keys()
+            .last()
+            .and_then(|&height| self.get_block_by_height(txn, height).transpose())
+            .transpose()
+    }
+
+    /// Returns vector blocks that satisfy the predicate, starting from the latest one and following
+    /// the ancestry chain.
+    fn get_blocks_until<F, Tx: Transaction>(
```

Comment:
I'd use `while` instead of `until` (in line with functions like `take_while`).

Reply:
Renamed in [`5035f0d` (#1216)](https://github.com/CasperLabs/casper-node/pull/1216/commits/5035f0d8a3087a306eb5e11a3b354995f7d8707b)